### ISSUE TITLE
Outdated price on taker leads to weird side effects

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#1327]
Version `0.4.0` mainnet umbrel

The position started at -2.0 % at a price of `38,869` - the current price is `37700`:

![image](https://user-images.githubusercontent.com/5557790/152238755-dc4f88e0-2245-4162-8bb3-f4045568152b.png)

When I open a new position I am immediately in the plus (see second position)

I think *some* price must be outdated for the taker.

**Comment by da-kami** [comment]
There we go, can't settle:

![image](https://user-images.githubusercontent.com/5557790/152239072-40ef9ddb-3364-4544-8e1e-8d0235b34a4b.png)


I hope this is related to my IP (and making too many connections to the BitMex price feed) and not an overall problem :/

**Comment by da-kami** [comment]
Restart of the app "fixes" it:

![image](https://user-images.githubusercontent.com/5557790/152239322-186d8425-7f0b-45ac-8c8f-218ddb884864.png)

**Comment by bonomat** [comment]
same problem on the maker: the UI shows an outdated price, meaning, the binary has an outdated quote somewhere

**Comment by bonomat** [comment]
The bitmex actor crashed and was restarted, ever since no new quote information: 

```
{"timestamp":"2022-02-02T14:11:27.42935635Z","level":"INFO","fields":{"message":"Actor stopped","actor":"bitmex_price_feed::Actor","reason":"Connection to BitMex API failed","restart":"true"},"target":"daemon::supervisor"}
```

```


{
  "timestamp": "2022-02-02T14:11:27.429400737Z",
  "level": "INFO",
  "fields": {
    "message": "Spawning new actor instance",
    "actor": "bitmex_price_feed::Actor"
  },
  "target": "daemon::supervisor"
}


{
  "timestamp": "2022-02-02T14:11:27.42935635Z",
  "level": "INFO",
  "fields": {
    "message": "Actor stopped",
    "actor": "bitmex_price_feed::Actor",
    "reason": "Connection to BitMex API failed",
    "restart": "true"
  },
  "target": "daemon::supervisor"
}
```

**Comment by thomaseizinger** [comment]
Do we get pings after that?

**Comment by bonomat** [comment]
> Do we get pings after that?

Tracing was not enabled for this

**Comment by da-kami** [comment]
I am running my umbrel taker on trace now to get more logs

**Comment by bonomat** [comment]
> I am running my umbrel taker on trace now to get more logs

enabled it on the maker as well

**Comment by bonomat** [comment]
Thought: given connection handling is hard, why don’t we just poll the http endpoint once a minute instead of using web sockets ?

**Comment by thomaseizinger** [comment]
> Thought: given connection handling is hard, why don’t we just poll the http endpoint once a minute instead of using web sockets ?

Yeah I guess with how the price feed actor evolved, we could certainly poll once a minute and store the latest price for others to retrieve it.

**Comment by bonomat** [comment]
fixed in #1342
